### PR TITLE
Precomp: Better error handling, hide cursor during print

### PR DESCRIPTION
**Pull request by ianshmean**
- Improves specificity of error handling ~~with a new `Base.PrecompileFailedException` type thrown by `Base.compilecache`. Dependent on https://github.com/JuliaLang/julia/pull/38109~~

- Hides cursor during pretty printing

### CODE REVIEW

File: src/API.jl [hunk 1]
```diff
@@ -1057,16 +1060,17 @@ function precompile(ctx::Context; internal_call::Bool=false, io::IO=stderr)
                 i += 1
                 wait(t)
             end
+            println(io, ansi_enablecursor)
```
Comment (KristofferC) [inline src/API.jl:1063]: Couldn't this be done in a `finally`?

Comment (ianshmean) [reply src/API.jl]: Indeed. Thanks

**Comment by KristofferC** [comment]
Could we update this for now not to depend on JuliaLang/julia#38109?

**Comment by ianshmean** [comment]
Done

**Comment by ianshmean** [comment]
Just a CI glitch on one of the MacOS machines

**Pull request merged**